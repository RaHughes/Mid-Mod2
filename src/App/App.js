import React, { Component } from 'react';
import './App.css';
import { CardContainer } from '../CardContainer/CardContainer.js';
import Form from '../Form/Form.js'


class App extends Component {
  constructor() {
    super()
    this.state = {
      cards:[]
    }
  }

  deleteData = id => {
    const del = {
      method: 'DELETE',
      headers: {
        'Content-Type': 'application/json'
      }
    }
    fetch(`http://localhost:3001/api/v1/reservations/${id}`, del)
    .then(res => console.log(res))
    .catch(err => console.log(err))
    this.fetchData()
    .then(data => this.setState({ cards: data }))
  }

  postData = obj => {
    const post = {
      method: 'POST',
      body: JSON.stringify(obj),
      headers: {
        'Content-Type': 'application/json'
      }
    }
    fetch('http://localhost:3001/api/v1/reservations', post)
    .then(res => console.log(res))
    .catch(err => console.log(err))
  }

  fetchData = () => {
    return fetch('http://localhost:3001/api/v1/reservations')
    .then(res => res.json())
  }

componentDidMount = () => {
  this.fetchData()
  .then(data => this.setState({ cards: data }))
}

sumbitReservation = res => {
  let reservation = { name: res.name, date: res.date, time: res.time, number: parseInt(res.number) }
  console.log(reservation)
  this.postData(reservation)
  this.setState({ cards: [...this.state.cards, reservation] })
}

  render() {
    return (
      <div className="App">
        <h1 className='app-title'>Turing Cafe Reservations</h1>
        <div className='resy-form'>
        <Form sumbitReservation={this.sumbitReservation}/>
        </div>
        <div className='resy-container'>
        <CardContainer cards={this.state.cards} deleteData={this.deleteData}/>
        </div>
      </div>
    )
  }
}

export default App;
