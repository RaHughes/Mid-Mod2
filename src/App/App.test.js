import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

describe('App', () => {
  it('Should match snapshot', () => {
    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  });
  it('Should update state when submitReservation is called', () => {
    const wrapper = shallow(<App />)
  })
})
