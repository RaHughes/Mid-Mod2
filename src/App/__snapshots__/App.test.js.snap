// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should match snapshot 1`] = `
<div
  className="App"
>
  <h1
    className="app-title"
  >
    Turing Cafe Reservations
  </h1>
  <div
    className="resy-form"
  >
    <Form
      sumbitReservation={[Function]}
    />
  </div>
  <div
    className="resy-container"
  >
    <CardContainer
      cards={Array []}
    />
  </div>
</div>
`;
